### ISSUE TITLE
PISP Support - Research FIDO authentication and align with our PISP work

### ISSUE THREAD

**Issue by godfreykutumela** [mojaloop/project#1830]
## **Goal**:
`As a` OSS CQS Workstream Contributor

`I want to` review the new FIDO Technical Authentication Standard
  
`so that` I can align our PISP FIDO implementation accordingly 

**Acceptance Criteria**:
- [ ] TBD

**Complexity**: Medium

**Uncertainty**: Medium

---

**Tasks**:
- [ ] TBD [ @godfreykutumela ]

**Done**
- [ ] Acceptance Criteria pass

## **Pull Requests**:
- [ ] TBD

## **Follow-up**:
- N/A

**Dependencies**:
- N/A

## **Accountability**:
- Owner: @godfreykutumela 
- QA/Review: @godfreykutumela

**Comment by godfreykutumela** [comment]
https://fidoalliance.org/technical-note-fido-authentication-and-emv-3-d-secure-using-fido-for-payment-authentication/

**Comment by godfreykutumela** [comment]
@lewisdaly @pedrosousabarreto and @rasputtintin, please make time to review and comment so that we can move this issue to the next stage

**Comment by lewisdaly** [comment]
Thanks Godfrey,

As Adrian mentioned in Slack, this standard doesn't really apply to us. 

The important takeways (at least from my perspective are):

The technical note is addressing a similar problem to what we face: FIDO authentication is between a user and a relying party (in our case the PISP is the relying party, in the paper it is the merchant). We want to be able to 'extend' FIDO so that a user can prove their identity and consent not only to the relying party, but to other interested parties (in our case the DFSP and Auth-Service).

This paper allows relying parties to expose some of the underlying FIDO artifacts to banks and card issuers, so that banks are satisfied that the authentication took place, which is perhaps an approach we can learn from.

**Comment by godfreykutumela** [comment]
Thanks, @lewisdaly for this closing comments.

**Comment by rasputtintin** [comment]
Hi @godfreykutumela I have completed doing some additional research around FIDO and 3D Secure looking at:
- 3Ds Authentication  Process and Mojaloop Security for PISP using FIDO
- Possible integrations
- Risks and concerns

Focus this week will be to collate the material into a presentable report.

**Comment by godfreykutumela** [comment]
@bukasaaime please upload the results of your analysis so that we can close this story

**Comment by bukasaaime** [comment]
Good day @godfreykutumela and @rasputtintin ,

I have looked at the following, regarding FIDO2 / WebAuthn and PISP: 

I have looked into the role that ML should play in the integration process. Should ML profivide end-points on behalf of the Payer FSP with regards to (1) user registration (2) user authentication (3) transaction authorization. 

1. PISP as a Relying Party. Would ML wrap PISP as a Relying Party or ML just be a passthrough for registration and auth.
2. PISP user registration process. Is user registration out of scope ? ML to provide end-point ? 
3. PISP user authentication process. ML to provide end-point ?
4. PISP transaction authorization process. ML to provide end-point ?
5. Authenticators and Attestation Certificates. Are the PISP users going to use authenticators other then the smart phone ?
6. Introduction of WebAuthn in case a browser is used to access PISP functions.
7. Review Credit Transaction flow. Is the Credit Transaction only Initiated after user has accepted the authorization request ?
8. Should FIDO / WebAuthn scope not be only limited to PISP and Payer FSP (outside of ML) and have Credit Transaction only initiated trough ML after the user has authorized (on the APP). What is the risk ?
